core/src/main/java/com/sap/ai/sdk/core/DestinationResolver.java

```diff
@@ -48,7 +55,29 @@ HttpDestination getDestination() {
             .onBehalfOf(TECHNICAL_USER_PROVIDER)
             .build();
 
-    return ServiceBindingDestinationLoader.defaultLoaderChain().getDestination(opts);
+    val destination = ServiceBindingDestinationLoader.defaultLoaderChain().getDestination(opts);
+    return setBasePath(addClientTypeHeader(destination));
+  }
+
+  @Nonnull
+  static HttpDestination fromCustomBaseDestination(@Nonnull final HttpDestination destination) {
+    // for custom base destinations we only add the client type header, since users are allowed to
+    // pass a custom base path
+    return addClientTypeHeader(destination);
+  }
+
+  @Nonnull
+  private static HttpDestination setBasePath(@Nonnull final HttpDestination destination) {
+    return DefaultHttpDestination.fromDestination(destination)
+        .uri(destination.getUri().resolve(BASE_PATH))
```

Comment:
The base path is intentionally absolute and will override any path given in the service binding. At least for now, the binding should never contain any path, so any given path would likely be faulty.

Reply:
> the binding should never contain any path

Feels like an undocumented assumption.
We don't know where the service binding is coming from.

Reply:
As long as users use a service binding produced by AI Core, this guaranteed.

Both approaches, merging paths as well as removing paths, can lead to confusion. But I think removing leads to less confusion, because it doesn't depend on user input, i.e. wether or not users add a trailing slash or not.

So yes, agree resolve can be confusing, I just hope this solution produces the least confusion 😄